### DISCUSSION TITLE
Clarification for the HelioVector function

### DISCUSSION THREAD

**Discussion by jffaust** [cosinekitty/astronomy#258] [Q&A] [answered]
Hello @cosinekitty ,

Looking at the docs for the [HelioVector](https://github.com/cosinekitty/astronomy/tree/master/source/js#HelioVector) function, it states that "**The position is not corrected for light travel time or aberration**".  At first glance, this led me to think that I should use another function (like GeoVector which corrects for light travel time and supports aberration correction) if I wanted to get the true/astrometric positions. 

I did find [your comment](https://github.com/cosinekitty/astronomy/discussions/241#discussioncomment-3628010) providing more details about these mechanics but I would still like some clarifications regarding the `HelioVector` function. 

- Q1: Does it return astrometric positions? 
- Q2: Why doesn't it correct for light travel time?
- Q3: Why doesn't it correct for aberration?

Thanks :)

**Comment by cosinekitty** [comment, marked as answer]
This is a great question. I'm glad I can clarify here, and hopefully it will help other people too.

Yes, HelioVector provides astrometric positions: it models a Newtonian "instantaneous" picture of the Solar System. It tells you where any body is, relative to the Sun, at the given time. The intention is that if you want to draw a picture of the Solar System as a whole, HelioVector might make more sense.

The GeoVector function tells you where an object *appears to be* at a given time as seen from the Earth, which is why GeoVector corrects for light travel time and, optionally, aberration also.

If you want to calculate how a target body appears from an observer body other than the Earth, there is the [BackdatePosition](https://github.com/cosinekitty/astronomy/tree/master/source/js#BackdatePosition) function, which corrects for light travel and optionally aberration, given any supported Solar System body as the vantage point. In fact, the call

```js
const vec = Astronomy.BackdatePosition(date, Astronomy.Body.Earth, body, aberration);
```

is equivalent to

```js
const vec = Astronomy.GeoVector(date, body, aberration);
```

If you want to correct for light travel for an arbitrary observer anywhere in space (not just from one of the major bodies), there is an even more general function [CorrectLightTravel](https://github.com/cosinekitty/astronomy/tree/master/source/js#CorrectLightTravel).

If you look at the code, you will see that GeoVector is implemented by calling BackdatePosition, and BackdatePosition calls CorrectLightTravel. The code for BackdatePosition can help serve as a reference for implementing arbitrary aberration correction also.

To calculate geocentric astrometric coordinates, you can call HelioVector twice, once for the Earth, and once for the target body, then subtract vectors: (geocentric body) = (heliocentric body) - (heliocentric Earth).

I hope this helps.

**Reply by cosinekitty** [reply]
I just realized my documentation generator doesn't do a very good job for [PositionFunction](https://github.com/cosinekitty/astronomy/tree/master/source/js#PositionFunction). The TypeScript declaration is

```typescript
/**
 * @brief A function for which to solve a light-travel time problem.
 *
 * The function {@link CorrectLightTravel} solves a generalized
 * problem of deducing how far in the past light must have left
 * a target object to be seen by an observer at a specified time.
 * This interface expresses an arbitrary position vector as
 * function of time that is passed to {@link CorrectLightTravel}.
 */
export abstract class PositionFunction {
    /**
     * @brief Returns a relative position vector for a given time.
     * @param {AstroTime} time
     *      The time at which to evaluate a relative position vector.
     */
    abstract Position(time: AstroTime): Vector;
}
```

**Reply by jffaust** [reply]
I understand the source of my confusion. 

"The GeoVector function tells you where an object appears to be at a given time as seen from the Earth, which is why GeoVector corrects for light travel time and, optionally, aberration also."

In my mind, correcting something means getting to the "true" value. When I look at the Sun in the sky, I have to correct it by considering the time it takes for light to travel to get its true position. For the GeoVector function however, it corrects the true position to get the apparent position for an observer on Earth. I guess it's all a matter of perspective :stuck_out_tongue: but maybe something we can clarify in the docs.
